src/engine/AddCombinedRowToTable.h

```diff
@@ -304,8 +306,8 @@ class AddCombinedRowToIdTable {
     // Note: There is quite some code duplication between this lambda and the
     // previous one. I have tried to unify them but this lead to template-heavy
     // code that was very hard to read for humans.
-    auto writeNonJoinColumn = [&result, oldSize, this]<bool isColFromLeft>(
-                                  size_t colIdx, size_t resultColIdx) {
+    auto writeNonJoinColumn = CPP_template_lambda(&result, oldSize, this)(
```

Comment:
To pass explicit values as template arguments, you can implement an
`ad_utility::ValueIdentity`  (same as `type_identity` ) but with a `::value`  instead of a 
` ::type` . Let me know if you need help.

Reply:
Done